### PR TITLE
optimize single attestation conversion

### PR DESCRIPTION
**Pull request by tbenr**
I run a profiler for ~4min on mainnet and I noticed this

<img width="847" height="592" alt="image" src="https://github.com/user-attachments/assets/67c23c9c-c110-4745-ae5a-ef844d46ebdf" />






We can avoid the serialization/deserialization of the singleAttestationAggregationBits when creating a the electra attestation (aggregated)


<img width="707" height="105" alt="image" src="https://github.com/user-attachments/assets/8af79251-8a2f-4677-9e48-8fb02e89e993" />




## Benchmark Results: SszBitlist Tree Construction Optimization                                                                                              
                                                                                                                                                             
  **Environment:** JDK 21.0.6, OpenJDK 64-Bit Server VM (Zulu), 1 fork, 5 warmup / 10 measurement iterations @ 1s each                                         
                                                          
  ### Throughput (ops/s, higher is better)                                                                                                                     
                                                                                                                                                               
  | Benchmark | Baseline | Optimized | Speedup |
  |---|---:|---:|---:|
  | `createBitlistAndSerialize` | 2,573,246 ± 22,197 | 11,401,721 ± 106,930 | **4.43x** |
  | `createBitlistAndHashTreeRoot` | 2,383,815 ± 33,825 | 8,053,888 ± 37,651 | **3.38x** |
  | `createBitlistAndOr` | 42,836,426 ± 163,337 | 40,525,583 ± 198,794 | ~1.0x |

  ### Allocations (B/op, lower is better)

  | Benchmark | Baseline | Optimized | Reduction |
  |---|---:|---:|---:|
  | `createBitlistAndSerialize` | 1,760 B/op | 928 B/op | **-47% (-832 B)** |
  | `createBitlistAndHashTreeRoot` | 1,720 B/op | 944 B/op | **-45% (-776 B)** |
  | `createBitlistAndOr` | 392 B/op | 392 B/op | 0% |

  ### Summary

  The optimization replaces the serialize/deserialize round-trip in `SszBitlistImpl` with direct tree construction from `BitSet` byte data. Benchmarks that
  force tree materialization (`sszSerialize()`, `hashTreeRoot()`) show **3.4-4.4x throughput improvement** and **~45-47% allocation reduction**. The `or()`
  benchmark is unaffected because it operates at the `BitlistImpl` level and defers tree construction (lazy supplier), confirming the optimization is correctly
   scoped.


## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> **Medium Risk**
> Touches SSZ tree construction for bitlists; incorrect byte sizing/padding could change hash tree roots or serialization for edge-case lengths, though new tests cover a range of sizes.
> 
> **Overview**
> **Optimizes `SszBitlist` tree materialization.** `SszBitlistImpl` no longer builds its backing tree via SSZ `serialize`/`deserialize`; it now constructs the tree directly from the underlying `BitSet` bytes via a new `SszBitlistSchemaImpl.createTreeFromBitData` path.
> 
> This adds `BitlistImpl.toByteArray()` as the raw bit source, updates tests to validate the new byte extraction and that direct tree construction matches the previous deserialization-based root, and extends JMH benchmarks to measure `create+serialize`, `create+or`, and `create+hashTreeRoot` scenarios.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6db4238fcf15820c18fafd6788d0e3f277bbdfe8. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

### CODE REVIEW

**Review by rolfyone — APPROVED** [review]
LGTM

File: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/impl/SszBitlistSchemaImpl.java [hunk 1]
```diff
@@ -70,6 +71,14 @@ public SszBitlist createFromElements(final List<? extends SszBit> elements) {
         IntStream.range(0, elements.size()).filter(i -> elements.get(i).get()).toArray());
   }
 
+  public TreeNode createTreeFromBitData(final int size, final byte[] bitSetBytes) {
+    final int dataByteLen = TreeUtil.bitsCeilToBytes(size);
+    final Bytes treeBytes =
+        dataByteLen == 0 ? Bytes.EMPTY : Bytes.wrap(Arrays.copyOf(bitSetBytes, dataByteLen));
+    final TreeNode dataTree = SchemaUtils.createTreeFromBytes(treeBytes, treeDepth());
+    return createTree(dataTree, size);
+  }
```
[the remark this answers is not in the document]
Comment (tbenr) [reply infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/impl/SszBitlistSchemaImpl.java:80]: we won't ever have 256 MB long bitlist!
